Trippy/API/UserAuthHandler.m

```diff
@@ -0,0 +1,37 @@
+//
+//  ParseHandler.m
+//  Trippy
+//
+//  Created by Catherine Lu on 7/7/22.
+//
+
+#import "UserAuthHandler.h"
+#import "Parse/Parse.h"
+
+@implementation UserAuthHandler
+
+- (void) logInWithUsername:(NSString *)username password:(NSString *)password {
+    [PFUser logInWithUsernameInBackground:username password:password block:^(PFUser * user, NSError * error) {
+        if (error) {
+            [self.delegate generalRequestFail:error];
+        } else {
+            [self.delegate loggedInSuccess];
+        }
```

Comment:
Why did you decide to go with delegates for this pattern? It's an interesting choice. One of the nice things with success/failure blocks is that there is inline context. You have a TODO on the failure method to make it more specific to the failure, which would happen by adding more methods to the delegate that your callers have to override and handle. If you used a block instead, the passed in block should know what you're trying to do and can handle the error appropriately.

Reply:
Thought it was one way to get around the long asynchronous nesting chains and break up the code a bit - that definitely makes sense about the blocks though, I mostly just wanted to try something new 😀